config/setupTests.ts

```diff
@@ -0,0 +1,12 @@
+import '@testing-library/jest-dom';
+
+jest.mock('react', () => {
+  const originalModule = jest.requireActual('react');
+
+  return {
+    __esModule: true,
+    ...originalModule,
+    useLayoutEffect: jest.requireActual('react').useEffect,
```

Comment:
I'm confused why you added this... can you give me some context? This mock feels unnecessary.

Reply:
There's a warning when using `useLayoutEffect` that is has no effect on server https://github.com/mui/material-ui/issues/15798 we are not using it anywhere (yet), but it helps to add it now so we don't have to deal with it later down the line.

Reply:
Interesting. Useful for sure... we might have utility components that could use this hook 🤔 

cc @vojtechszocs -- thoughts?

Reply:
@karelhala This addresses the issue when testing a component that calls the `React.useLayoutEffect` hook, right?

Reply:
That is correct @vojtechszocs

Reply:
@andrewballantyne I'm fine keeping this bit, just want to make sure the provided override (`React.useEffect`) is fully compatible with what we're replacing (`React.useLayoutEffect`) in terms of both signature and behavior.

Reply:
It's noted as such on the React Docs

> useLayoutEffect... The signature is identical to useEffect...
https://reactjs.org/docs/hooks-reference.html#uselayouteffect